src/plugProjectOgawaU/ogScreen.cpp

```diff
@@ -1,1880 +1,461 @@
+
+	return maskedTag;
 }
 
 /*
  * --INFO--
  * Address:	80302920
  * Size:	0000A0
  */
-u64 maskTag(u64, u16, u16)
+u64 maskTag(u64 tag, u16 num, u16 mask)
 {
-	/*
-stwu     r1, -0x20(r1)
-mflr     r0
-clrlwi   r5, r5, 0x10
-stw      r0, 0x24(r1)
-cmplwi   r5, 8
-stmw     r27, 0xc(r1)
-mr       r28, r3
-mr       r27, r4
-mr       r29, r6
-blt      lbl_80302954
-li       r4, 0
-li       r3, 0
-b        lbl_803029AC
-
-lbl_80302954:
-addi     r0, r5, -1
-li       r3, 0
-rlwinm   r31, r0, 3, 0x10, 0x1c
-li       r4, 0xff
-mr       r5, r31
-bl       __shl2i
-clrlwi   r5, r29, 0x10
-nor      r29, r4, r4
-cmplwi   r5, 9
-nor      r30, r3, r3
-bgt      lbl_8030298C
-addi     r4, r5, 0x30
-srawi    r3, r4, 0x1f
-b        lbl_80302994
-
-lbl_8030298C:
-addi     r4, r5, 0x37
-srawi    r3, r4, 0x1f
-
-lbl_80302994:
-mr       r5, r31
-bl       __shl2i
-and      r5, r27, r29
-and      r0, r28, r30
-or       r4, r4, r5
-or       r3, r3, r0
-
-lbl_803029AC:
-lmw      r27, 0xc(r1)
-lwz      r0, 0x24(r1)
-mtlr     r0
-addi     r1, r1, 0x20
-blr
-	*/
+	if (num >= 8) {
+		return 0;
+	}
+
+	int shift       = ((num - 1) * 8) & 0xFFF8;
+	u64 baseChar    = 0xFF;
+	u64 extractChar = ~(baseChar << shift);
+
+	u64 maskedChar;
+	if (mask <= 9) {
+		maskedChar = mask + 0x30;
+	} else {
+		maskedChar = mask + 0x37;
+	}
+
+	return (maskedChar << shift) | (tag & extractChar);
 }
 
+const char* PikiIconTextureName[]
+    = { "toumei_piki.bti", "bp_l64.bti", "bp_b64.bti", "bp_f64.bti",  "rp_l64.bti",  "rp_b64.bti",  "rp_f64.bti",
+	    "yp_l64.bti",      "yp_b64.bti", "yp_f64.bti", "blp_l64.bti", "blp_b64.bti", "blp_f64.bti", "wp_l64.bti",
+	    "wp_b64.bti",      "wp_f64.bti", "cha_l.bti",  "cha_b.bti",   "cha_f.bti" };
+
 /*
  * --INFO--
  * Address:	803029C0
  * Size:	0000B4
  */
-i32 CalcKeta(u32)
+u16 CalcKeta(u32 p1)
 {
-	/*
-stwu     r1, -0x40(r1)
-mflr     r0
-stw      r0, 0x44(r1)
-stfd     f31, 0x38(r1)
-lfd      f31, lbl_8051D528@sda21(r2)
-stfd     f30, 0x30(r1)
-lfd      f30, lbl_8051D550@sda21(r2)
-stw      r31, 0x2c(r1)
-lis      r31, 0x4330
-stw      r30, 0x28(r1)
-li       r30, 1
-stw      r29, 0x24(r1)
-li       r29, 1
-stw      r28, 0x20(r1)
-mr       r28, r3
-
-lbl_803029FC:
-xoris    r0, r29, 0x8000
-stw      r31, 8(r1)
-lfd      f1, lbl_8051D548@sda21(r2)
-stw      r0, 0xc(r1)
-lfd      f0, 8(r1)
-fsub     f2, f0, f30
-bl       pow
-stw      r28, 0x14(r1)
-stw      r31, 0x10(r1)
-lfd      f0, 0x10(r1)
-fsub     f0, f0, f31
-fcmpo    cr0, f0, f1
-cror     2, 1, 2
-bne      lbl_80302A48
-addi     r0, r29, 1
-addi     r29, r29, 1
-cmpwi    r29, 0xa
-clrlwi   r30, r0, 0x10
-blt      lbl_803029FC
-
-lbl_80302A48:
-lwz      r0, 0x44(r1)
-mr       r3, r30
-lfd      f31, 0x38(r1)
-lfd      f30, 0x30(r1)
-lwz      r31, 0x2c(r1)
-lwz      r30, 0x28(r1)
-lwz      r29, 0x24(r1)
-lwz      r28, 0x20(r1)
-mtlr     r0
-addi     r1, r1, 0x40
-blr
-	*/
+	u16 keta = 1;
+	for (int i = 1; i < 10; i++) {
+		if (p1 >= pow(10.0, (double)i)) {
+			short j = i + 1;
+			keta    = j;
```

Comment:
Any reason to declare `j` here? `keta = i + 1;` has the same codegen: https://decomp.me/scratch/45TKK

Reply:
probably a hangover from sorting out type issues - will remove it